scipy/interpolate/_isotonic.py

```diff
@@ -0,0 +1,123 @@
+from __future__ import annotations
+from typing import TYPE_CHECKING
+
+import numpy as np
+
+from scipy.optimize import isotonic_regression
+
+if TYPE_CHECKING:
+    import numpy.typing as npt
+
+
+__all__ = ["IsotonicInterpolator"]
+
+
+# TODO: Should this inherit from _Interpolator1D?
```

Comment:
Not sure if this would bring anything of value. If anything, new additions are best walk and quack like a PchipInterpolator. I actually wonder if this thing should just construct a PPoly or BSpline instance.

Reply:
You mean a PPoly od BSpline of polynomial degree 0, right?

Reply:
Yes, a degree-1 piecewise polyniomial in either basis, whichever is more convenient. This way you get things like `.solve` for free.

Note a suggestion to instead just document the use of `np.interp` below. Not everything needs to be an object.